### PR TITLE
Bugfix Nodes Table Merging

### PR DESCRIPTION
**Pull request by bburns632**
# The Issue
Node level statistics are being merged out of order causing issue #165.  

As currently written, node statistics are merged under this paradigm: 
``` 
            outNodeDT <- self$nodes

            outDegreeResult <- igraph::centralization.degree(
                graph = pkg_graph
                , mode = "out"
            )

            # update data.tables
            outNodeDT[, outDegree := outDegreeResult[['res']]] # nodes
```

However, `pkg_graph` (which drives the order of outDegreeResult) and the nodes table, `outNodeDT`, have different node orders: 

```
cbind(igraph::vertex.attributes(pkg_graph)$name
+       , outNodeDT$node)
     [,1]            [,2]           
[1,] "baseballstats" "baseballstats"
[2,] "base"          "methods"      
[3,] "methods"       "stats"        
[4,] "utils"         "graphics"     
[5,] "stats"         "base"         
[6,] "grDevices"     "utils"        
[7,] "graphics"      "grDevices"    
```
This causes the mismatch seen in #165.  

# The Fix
Utilizing the existing method `update_nodes` and some minor manipulation fixes this issue. 

# The Tests So It is Not Merged Again
Units tests have been added to: 
1. confirm all expected statistics are calculated
2. For one external package dependency (`stats` in `baseballstats`), confirm _node level_ statistics are calculated and assigned _correctly_
3. For one internal function dependency (`slugging_avg` in `baseballstats`), confirm  _node level_ statistics are calculated and assigned _correctly_
4. For one object (`KingOfTheEarth ` in `milne`), confirm  _node level_ statistics are calculated and assigned _correctly_
5. For external package dependency network, confirm _network level_ statistics are calculated and assigned _correctly_
6. For internal function dependency network, confirm _network level_ statistics are calculated and assigned _correctly_
7. For object inheritance network,  confirm _network level_ statistics are calculated and assigned _correctly_

Note, changes to these unit tests will be required when: 
* `baseballstats` or  `milne` is changed
* a node level statistic is added
* a network level statistic is added
* igraph changes any of their functions affecting the calculations (unlikely)

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+# pkgnet 0.3.1.9999 (dev)
+
+## NEW FEATURES
+None
+
+## CHANGES
+* Added unit tests for network measure calculations.
```
Comment (jameslamb) [inline NEWS.md]: can you add an issue link on this one too?

File: R/AbstractGraphReporter.R [hunk 2]
```diff
@@ -131,9 +132,14 @@ AbstractGraphReporter <- R6::R6Class(
                 graph = pkg_graph
                 , directed = TRUE
             )
-
+            
+            outBetweenessResultDT <- data.table::data.table(node = igraph::vertex.attributes(pkg_graph)[['name']]
```
Comment (jameslamb) [inline R/AbstractGraphReporter.R]: can you add `#' importFrom igraph vertex.attributes`?

File: R/AbstractGraphReporter.R [hunk 3]
```diff
@@ -131,9 +132,14 @@ AbstractGraphReporter <- R6::R6Class(
...
+                                                        , outBetweeness = outBetweenessResult[['res']]
+            )
+            
             # update data.tables
-            outNodeDT[, outBetweeness := outBetweenessResult$res] # nodes
```
Comment (jameslamb) [inline R/AbstractGraphReporter.R]: good change!

File: tests/testthat/test-NetworkMeasures.R [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+rm(list = ls())
+# Configure logger (suppress all logs in testing)
+# expect_silents only work with this logger turned off; only alerts with warnings
+loggerOptions <- futile.logger::logger.options()
+if (!identical(loggerOptions, list())){
+    origLogThreshold <- loggerOptions[[1]][['threshold']]
+} else {
+    origLogThreshold <- futile.logger::INFO
+}
+futile.logger::flog.threshold(0)
```
Comment (jameslamb) [inline tests/testthat/test-NetworkMeasures.R]: for another day...now that you taught me that `setupTestEnv` thing...could we put a `silence_logger` function in that that we reused in every test? Or even just run this code there?

I'll make an issue and we can discuss there

Comment (jameslamb) [reply tests/testthat/test-NetworkMeasures.R]: created #170

File: tests/testthat/test-NetworkMeasures.R [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+# note, does not test order
+test_that('All expected Node Level Network Measures are calculated', {
+    
+    # External Package Dependencies
+    expect_true(object = all(expectedNodeLevelMeasures %in% names(extDepNodes))
+                , info = 'Network Measures for external package dependencies'
+                )
+    
+    # Internal Function Dependencies
+    expect_true(object = all(expectedNodeLevelMeasures %in% names(intFuncNodes))
```
Comment (jameslamb) [inline tests/testthat/test-NetworkMeasures.R]: @bburns632 you are the only person I've ever seen use the keyword arg `object = ` in `testthat` code. Is there are reason you like it over this format?

```
expect_true('Network Measures for external package dependencies', {
    all(expectedNodeLevelMeasures %in% names(extDepNodes))
})
```

Not trying to passive-aggressively recommend you change, honestly curious.

Comment (bburns632) [reply tests/testthat/test-NetworkMeasures.R]: Just following the docs.  What you wrote above looks like the format for a [test_that()](https://testthat.r-lib.org/reference/test_that.html) call, not an [expect_true()](https://testthat.r-lib.org/reference/logical-expectations.html) call.

File: tests/testthat/test-NetworkMeasures.R (hunk 4 above)
Comment (bburns632) [reply tests/testthat/test-NetworkMeasures.R]: Honestly, I just copy-pasta'd the header and footer code from the other test files.  Open to changes after this PR.

**Comment by bburns632** [comment]
@jayqi I'm looking forward to that upcoming refactor.  It looks clean and probably easier to test.  Thanks for holding off for the moment.

**Pull request merged**